src/query/functions/src/scalars/datetime.rs

```diff
@@ -131,6 +135,109 @@ fn int64_domain_to_timestamp_domain<T: AsPrimitive<i64>>(
     })
 }
 
+fn register_convert_timezone(registry: &mut FunctionRegistry) {
+    // 2 arguments function [target_timezone, src_timestamp]
+    registry.register_passthrough_nullable_2_arg::<StringType, TimestampType, TimestampType, _, _>(
+        "convert_timezone",
+        |_, _, _| FunctionDomain::MayThrow,
+        eval_convert_timezone,
+    );
+
+    // 3 arguments function [src_timezone, target_timezone, src_timestamp]
+    registry.register_passthrough_nullable_3_arg::<StringType, StringType, TimestampType, TimestampType, _, _>(
+        "convert_timezone",
+        |_, _, _, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_3_arg::<StringType, StringType, TimestampType, TimestampType>(
+            |src_tz, target_tz, src_timestamp, output, ctx| {
+                // Parsing parameters
+                let t_tz: Tz = match target_tz.parse() {
+                    Ok(tz) => tz,
+                    Err(e) => {
+                        return ctx.set_error(
+                            output.len(),
+                            format!("cannot parse target `timezone`. {}", e),
+                        );
+                    }
+                };
+                let s_tz: Tz = match src_tz.parse() {
+                    Ok(tz) => tz,
+                    Err(e) => {
+                        return ctx.set_error(
+                            output.len(),
+                            format!("cannot parse src `timezone`. {}", e),
+                        );
+                    }
+                };
+
+                let p_src_timestamp: i64 = match Some(src_timestamp){
+                    Ok(timestamp) => {
+                        timestamp.Utc.timestamp_opt(src_timestamp, 0).unwrap();
+                    },
+                    Err(e) => {
+                        return ctx.set_error(
+                            output.len(),
+                            format!("cannot parse target `timezone`. {}", e),
+                        );
+                    }
+                };
+
+
+                // Create dummy timezone
+                let utc_now: DateTime<Utc> = Utc::now();
```

Comment:
I think no need to create dummy timezone. You can use this https://docs.rs/chrono/latest/chrono/offset/trait.TimeZone.html#method.timestamp_opt to convert ts to source timezone.

Then convert the ts with src tz to target tz.

Reply:
e.g. https://github.com/datafuselabs/databend/blob/78d41aedc7f078485ee4ada10281d9f81a09caa5/src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs#L194